runtime/src/main/java/org/corfudb/runtime/LogReplicationLogicalGroupClient.java

```diff
@@ -0,0 +1,393 @@
+                        log.warn(String.format("Record not found for group: %s", logicalGroup));
+                    }
+
+                    txn.commit();
+                    return null;
+                } catch (TransactionAbortedException tae) {
+                    log.error(String.format("[%s] Unable to remove destination, will be retried.", clientName), tae);
+                    throw new RetryNeededException();
+                }
+            }).run();
+        } catch (InterruptedException e) {
+            log.error(String.format("[%s] Unable to remove destination.", clientName), e);
+            throw new UnrecoverableCorfuInterruptedError(e);
+        }
+    }
+
+    /**
+     * Remove a list of destinations from the LogReplicationModelMetadataTable for a specific group.
+     *
+     * @param logicalGroup The group of tables that are replicated, represented as a string.
+     * @param remoteDestinations The clusters which should be removed from the logical group for all future replication.
+     */
+    public void removeDestination(String logicalGroup, List<String> remoteDestinations) {
+        Preconditions.checkArgument(isValid(logicalGroup), String.format("[%s] %s", clientName, "logicalGroup is null or empty."));
+        Preconditions.checkArgument(isValid(remoteDestinations), String.format("[%s] %s", clientName, "remoteDestinations is null or empty."));
+        Preconditions.checkArgument(hasNoNullOrEmptyElements(remoteDestinations), String.format("[%s] %s", clientName, "remoteDestinations contains null or empty elements."));
+        Set<String> removeRemoteDestinationsSet = deduplicate(remoteDestinations);
+        try {
+            IRetry.build(ExponentialBackoffRetry.class, () -> {
+                try (TxnContext txn = corfuStore.txn(CORFU_SYSTEM_NAMESPACE)) {
+                    ClientDestinationInfoKey clientInfoKey = buildClientDestinationInfoKey(logicalGroup);
+                    DestinationInfoVal clientDestinations = txn.getRecord(sourceMetadataTable, clientInfoKey).getPayload();
+
+                    if (clientDestinations != null) {
+                        Set<String> clientDestinationIdsSet = new HashSet<>(clientDestinations.getDestinationIdsList());
+                        List<String> clientDestinationIdsList = clientDestinationIdsSet.stream()
+                                .filter(i -> !removeRemoteDestinationsSet.contains(i))
+                                .collect(Collectors.toList());
+
+                        if (clientDestinationIdsList.size() < clientDestinationIdsSet.size()) {
+                            if (clientDestinationIdsList.size() + removeRemoteDestinationsSet.size() > clientDestinationIdsSet.size()) {
+                                removeRemoteDestinationsSet.removeAll(clientDestinationIdsSet);
+                                log.info(String.format("[%s] Following destinations to remove are not present: %s", clientName, removeRemoteDestinationsSet));
```

Comment:
actually, this message will always get printed even if there are no remaining ids in removeRemoteDestinationsSet.  It will also print the set reference and not the elements themselves.

I'd suggest adding a check if the set is empty.  If not, printing the list of remaining destinations.